library/src/main/java/com/novoda/downloadmanager/DownloadManagerBuilder.java

```diff
@@ -223,7 +223,7 @@ public void onServiceDisconnected(ComponentName name) {
 
         Optional<NotificationChannel> notificationChannel = notificationChannelCreator.createNotificationChannel();
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O && notificationChannel.isPresent()) {
```

Comment:
I guess it's not really for this pull request, but it looks like we're doing this check twice - once inside the `createNotificationChannel` method and once here.

Reply:
We need this check twice 😢 the one inside `createNotificationChannel` can be defined by the client, not by the library, so we need this safety 😬

Reply:
I don't think that means we need the check twice. I'm going to do some digging / refactoring / exploration